### PR TITLE
high contrast themes support

### LINKED ISSUE #41: add support for high contrast themes
High contrast themes, such as those with a black background, are not currently supported. This causes inlinehints to blend into the background, for example: "exported from" in modulite.

### PR DESCRIPTION
**Pull request by Hidanio**

### CODE REVIEW

File: src/main/kotlin/com/vk/modulite/highlighting/hints/BaseInlayHintsCollector.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ abstract class BaseInlayHintsCollector(val project: Project, editor: Editor) : F
     }
 
     private fun singleModuliteNamePresentation(name: String, nameToShow: String): InlayPresentation {
-        val textPresentation = factory.text(nameToShow)
+        val textPresentation = factory.smallTextWithoutBackground(nameToShow)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/modulite/highlighting/hints/BaseInlayHintsCollector.kt:75]: I disagree.
The [smallTextWithoutBackground](https://github.com/JetBrains/intellij-community/blob/7a1566b145b677302a256e9c4ed44e452e526443/platform/lang-impl/src/com/intellij/codeInsight/hints/presentation/PresentationFactory.kt#L54) and [text](https://github.com/JetBrains/intellij-community/blob/7a1566b145b677302a256e9c4ed44e452e526443/platform/lang-impl/src/com/intellij/codeInsight/hints/presentation/PresentationFactory.kt#L81) functions are organized differently.

It still seems to me that it is possible to use `container`. Or I'll make my own `textWithoutBackground`.

File: src/main/kotlin/com/vk/modulite/highlighting/hints/PhpInlayHintsCollector.kt [hunk 2]
```diff
@@ -137,7 +137,7 @@ class PhpInlayHintsCollector(
             ""
         }
 
-        val visibilityPresentation = factory.text(visibility)
+        val visibilityPresentation = factory.smallTextWithoutBackground(visibility)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/modulite/highlighting/hints/PhpInlayHintsCollector.kt:140]: And then

File: src/main/kotlin/com/vk/modulite/highlighting/hints/BaseInlayHintsCollector.kt (hunk 1 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/modulite/highlighting/hints/BaseInlayHintsCollector.kt:75]: Yes, they are different and that why we are using it. `container` had no parameters for foreground color (for example). In [source code](https://github.com/JetBrains/intellij-community/blob/c856c6ac330e884c7425f95e8d8396aae45b572f/platform/lang-impl/src/com/intellij/codeInsight/hints/presentation/WithAttributesPresentation.kt#L45) of the `WithAttributesPresentation` we can see that flag `DefaultLanguageHighlighterColors.INLAY_TEXT_WITHOUT_BACKGROUND` will apply custom settings for display. In this context we have inlay hints  and similar flag for this. I did not find any side effects for this. `container` can be used for manipulating around /object/, but not properties (in this context - text).
We can create our own `textWithoutBackground`, but why do we need it?
As i understand your idea, you will create `TextAttributes`, `OwnTextWithoutBackground` and create instance of `WithAttributesPresentation`.  I think that it should be the similar with standart `smallTextWithoutBackground`

Comment (Danil42Russia) [reply src/main/kotlin/com/vk/modulite/highlighting/hints/BaseInlayHintsCollector.kt:75]: Decided it was time-consuming

**Review by Danil42Russia — DISMISSED** [review]

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**